server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/migration/MappingsSourcesMigration.java

```diff
@@ -28,21 +31,56 @@
 import org.apache.james.rrt.lib.Mapping;
 import org.apache.james.rrt.lib.MappingSource;
 import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
 import reactor.core.publisher.Mono;
 
 public class MappingsSourcesMigration implements Migration {
     private static final Logger LOGGER = LoggerFactory.getLogger(MappingsSourcesMigration.class);
+    private static final String TYPE = "mappingsSourcesMigration";
+
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private final AtomicLong successfulMappingsCount;
+        private final AtomicLong errorMappingsCount;
+
+        AdditionalInformation() {
+            this.successfulMappingsCount = new AtomicLong(0);
+            this.errorMappingsCount = new AtomicLong(0);
+        }
+
+        public long getSuccessfulMappingsCount() {
+            return successfulMappingsCount.get();
+        }
+
+        public long getErrorMappingsCount() {
+            return errorMappingsCount.get();
+        }
+
+        @JsonIgnore
+        void incrementSuccessfulCount() {
```

Comment:
`AdditionalInformation` should not be responsible of updating progress.

The Task should have this responsibility

AdditionalInformation is just here for exposing task informations, not managing them.

This also means `successfulMappingsCount` and `errorMappingsCount` should be task parameters.